app/controllers/password_resets_controller.rb

```diff
@@ -53,17 +53,21 @@ def update
     @password_reset.user.password_confirmation = params[:password_confirmation]
     @password_reset.user.validate_password = true
 
-    unless @password_reset.user.save
+    begin
+      User.transaction do
+        @password_reset.user.save!
+        @password_reset.update!(used: true)
+      end
+      flash[:success] = "Password successfully changed."
+      redirect_to root_url
+    rescue ActiveRecord::RecordInvalid
       flash.now[:error] = {}
       flash.now[:error][:message] = "Could not update password."
       flash.now[:error][:array] = @password_reset.user.errors.full_messages
+      flash.now[:error][:array] = @password_reset.errors.full_messages
```

Comment:
You want a `+=` here, not `=` (otherwise there's a useless assignment on the line above)